manifests/init.pp

```diff
@@ -229,53 +254,53 @@
+  Enum['file', 'string'] $authkey_source                                      = $::corosync::params::authkey_source,
+  String $authkey                                                             = $::corosync::params::authkey,
+  Optional[Integer] $threads                                                  = $::corosync::params::threads,
+  String $port                                                                = $::corosync::params::port,
+  Variant[String, Array] $bind_address                                        = $::corosync::params::bind_address,
+  Optional[String] $multicast_address                                         = $::corosync::params::multicast_address,
+  Optional[Array] $unicast_addresses                                          = $::corosync::params::unicast_addresses,
+  Boolean $force_online                                                       = $::corosync::params::force_online,
+  Boolean $check_standby                                                      = $::corosync::params::check_standby,
+  Boolean $log_timestamp                                                      = $::corosync::params::log_timestamp,
+  Boolean $log_file                                                           = $::corosync::params::log_file,
+  Optional[Stdlib::Absolutepath] $log_file_name                               = $::corosync::params::log_file_name,
+  Boolean $debug                                                              = $::corosync::params::debug,
+  Boolean $log_stderr                                                         = $::corosync::params::log_stderr,
+  Enum['debug', 'info', 'notice', 'warning', 'err', 'emerg'] $syslog_priority = $::corosync::params::syslog_priority,
+  Boolean $log_function_name                                                  = $::corosync::params::log_function_name,
+  Optional[Enum['none', 'active', 'passive']] $rrp_mode                       = $::corosync::params::rrp_mode,
+  Optional[Integer] $netmtu                                                   = $::corosync::params::netmtu,
+  Optional[Integer[0,255]] $ttl                                               = $::corosync::params::ttl,
+  Optional[Enum['ykd', 'none']] $vsftype                                      = $::corosync::params::vsftype,
+  Boolean $package_corosync                                                   = $::corosync::params::package_corosync,
+  Boolean $package_crmsh                                                      = $::corosync::params::package_crmsh,
+  Boolean $package_pacemaker                                                  = $::corosync::params::package_pacemaker,
+  Boolean $package_pcs                                                        = $::corosync::params::package_pcs,
+  Optional[Array] $packageopts_corosync                                       = $::corosync::params::package_install_options,
```

Comment:
From what I understand, it's better to put the undef directly here when it's an optional param.

Reply:
It's not undef, if used with Debian 8:
```
Optional[Array] $packageopts_corosync = $::corosync::params::package_install_options
```
```
if versioncmp($::operatingsystemrelease, '8') >= 0 {                  
...
             $package_install_options = ['-t', 'jessie-backports']       
```
So setting it `undef` would unset `['-t', 'jessie-backports']` on Debian 8 machines which don't use this explicitly. Or do I miss something?